### DISCUSSION TITLE
msg response

### DISCUSSION THREAD

**Discussion by jagonzalezga** [eModbus/eModbus#195] [Q&A] [answered]
hello, good day, I need your help to obtain the message data, my response contains 31 response data, so I want to obtain that data and save it in a data array, could someone help me with a simple example to obtain data , I rely on the following function:
`void handleData(ModbusMessage response, uint32_t token) 
{
  Serial.printf("Response: serverID=%d, FC=%d, Token=%08X, length=%d\n", response.getServerID(), response.getFunctionCode(), token, response.size());
  for (auto& byte1 : response) 
  {
    Serial.printf("%02X ", byte1);
}
}`

what is the correct way to use " [ ] " operator?, because I enter it into an array but I think the memory is full but I don't know the correct way to delete the data and fill the data array again

**Comment by Miq1** [comment, marked as answer]
Provided your data is 31 values of type ``uint16_t``, you could do something like
```
...
// Global data array to hold the received data
const uint8_t SLOTS(31);
uint16_t myArray[SLOTS];  
...
void handleData(ModbusMessage response, uint32_t token) {
  // you should check for the correct function code and data length here!
  uint16_t position = 3;  // first data right after the length byte 
  for (uint8_t arrayIndex = 0; arrayIndex < SLOTS; arrayIndex++) { 
    position = response.get(position, myArray[arrayIndex]); // get next value, advancing the byte position
  } 
}
```
Note that I wrote that down without testing it, so there may be typos in it.

**Reply by jagonzalezga** [reply]
thanks for your help Miq1, it helped me.
